rfcs/text/testing-refinement.md

```diff
@@ -0,0 +1,135 @@
+
+If we run a visual regression test on a pull request from a 3rd party, they will be able to see the Applitools pass/fail test results but they will not be able to access the URL to the Applitools dashboard to view the full details. Where we are on one hand encouraging collaboration from other areas within Sage and beyond and expecting them to ultimately test their own improvements and fixes, this will limit how far they can test as it will still require us to investigate and report their visual testing failures on their behalf.
+
+# Alternatives
+
+There is some duplication between the Cypress regression tests and the unit tests. Indeed, the act of setting knobs to various states to set a prerequisite state and the result asserted on is already done in many of the unit tests. An exercise to remove duplicate tests in the regression suite could be carried out to speed up the run time. The recent removal of support of the ‘classic’ theme and related feature files reduced the regression suite run time by 1-1.5 hours. We could upgrade our Cypress package to a paid for service which would allow the tests to be parallelised rather than them all running in one thread, reducing the run time further. All of these actions though would not reduce the regression suite run time by the required factor nor provide the visual comparison testing capability that Applitools integration would.
+
+One advantage of integrating Applitools is having the option of rendering the DOM snapshots in many other operating system, browser, device and viewport configurations we like by using their Ultrafast Grid. This does not mean though that we are testing our components in these other configurations at the time of test execution. There are other vendors, including CrossBrowserTesting and BrowserStack, which would allow us to run our required tests directly in these other configurations of real machines. However, this would mean a change of automation framework from Cypress to Selenium. This seems like a drastic change at this point as all the files, including the support files like helper files, indexes, locators etc, would need rewriting. In addition, not all of these other products are free for open source repositories as Applitools is and don’t all provide a dashboard which is as user-friendly as that of Applitools.
```

Comment:
I don't think we will change our framework for cypress to selenium

Reply:
Agreed, at least for the foreseeable future. I had to list it as an alternative though.